app/src/main/java/org/sagebionetworks/bridge/researchstack/CrfTaskFactory.java

```diff
@@ -315,4 +339,76 @@ public void startSettingsScreen(Context context, String externalID, String versi
         context.startActivity(IntentFactory.INSTANCE.newTaskIntent(
                 context, CrfSettingsActivity.class, task, taskResult));
     }
+
+    private CrfFormStep createCrfFormStep(Context context, CompoundQuestionSurveyItem item) {
+        if (item.items == null || item.items.isEmpty()) {
+            throw new IllegalStateException("compound surveys must have step items to proceed");
+        }
+
+        List<QuestionStep> questionSteps = new ArrayList<>();
+        for (SurveyItem subItem : item.items) {
+            if (subItem.getCustomTypeValue() != null &&
+                    subItem.getCustomTypeValue().equals(CrfSurveyItemAdapter.CRF_INTEGER_SURVEY_ITEM_TYPE)) {
+                QuestionStep step = createCustomIntegerBody(context, (QuestionSurveyItem) subItem);
+                questionSteps.add(step);
+            } else if (subItem.getCustomTypeValue() != null &&
+                    subItem.getCustomTypeValue().equals(CrfSurveyItemAdapter.CRF_SINGLE_CHOICE_SURVEY_ITEM_TYPE)) {
+                QuestionStep step = createCustomSingleChoiceBody(context, (QuestionSurveyItem) subItem);
+                questionSteps.add(step);
+            } else if (subItem instanceof QuestionSurveyItem) {
+                QuestionStep step = createQuestionStep(context, (QuestionSurveyItem)subItem);
+                questionSteps.add(step);
+            }
+        }
+
+        CrfFormStep step = new CrfFormStep(item.identifier, item.title, item.text, questionSteps);
+        step.setAutoFocusFirstEditText(true);
+
+        step.setSkipIfPassed(item.skipIfPassed);
+        step.setSkipToStepIdentifier(item.skipIdentifier);
+
+
+        fillQuestionStep(item, step);
+        if (item.expectedAnswer != null) {
+            step.setExpectedAnswer(item.expectedAnswer);
+        }
+        if (item.skipTitle != null) {
+            step.setSkipTitle(item.skipTitle);
+            // we can assume that if we set the skip title, we want to show the skip button
+            step.setOptional(true);
+        }
+
+        return step;
+    }
+
+    public QuestionStep createCustomIntegerBody(Context context, QuestionSurveyItem item) {
+        AnswerFormat format = null;
+        if (!(item instanceof IntegerRangeSurveyItem)) {
+            throw new IllegalStateException("Error in json parsing, QUESTION_INTEGER types must be IntegerRangeSurveyItem");
+        }
+        IntegerRangeSurveyItem intItem = (IntegerRangeSurveyItem)item;
+        int min = (intItem.min == null) ? 0 : intItem.min;
```

Comment:
Create a "fillIntegerSurveyItem" method?  I like to re-use the base code so we don't end up having to fix bugs in multiple locations.  I guess same goes for the single choice and the question steps within the form step.